.mergify.yml

```diff
@@ -0,0 +1,23 @@
+queue_rules:
+  # This rule is for https://docs.mergify.com/commands/queue/
+  # and can be triggered with: @mergifyio queue
+  - name: default
+    merge_conditions:
+      # all github action checks in this list are required to merge a pull request
+      - check-success=Attributes
+      - check-success=Check
+      - check-success=Outpaths (aarch64-darwin)
+      - check-success=Outpaths (aarch64-linux)
+      - check-success=Outpaths (x86_64-darwin)
+      - check-success=Outpaths (x86_64-linux)
+      - check-success=Process
+      - check-success=Request
+      - check-success=Tag
+      - check-success=editorconfig-check
+      - check-success=label-pr
+      - check-success=nix-files-parseable-check
+      - check-success=nixfmt-check
+      - check-success=nixpkgs-vet
+    merge_method: rebase
```

Comment:
Why rebase? Rebasing gets rid of the signature and recreates the commit with entirely new hash.

Reply:
It makes the git history more readable. I honestly don't see much security win from having those signatures unless we make them mandatory for everyone and also have a way to verify that someones actually owns this ssh key (i.e. keysigning in person). We already see from which account a change came from, if someones github account gets compromised, the attacker might as well just upload a new ssh key. There are a lot of cases in practice already where we amend commits from contributors or rebase on their behalf, so signatures get removed already.
It would be more interesting if we had signatures for merging pull request, but since we use exclusively the "merge" button for that, it will be always the github signing key anyway. Which brings me back to my initial point, we need to trust github and that our users don't loose control over their account.

Reply:
I used to prefer rebase merge as well, as that would get rid of the merge commit, but moved away from doing that, since most people on Nixpkgs seemed to prefer "preserving the git history" by keeping the old commit hash (no rebase merge).

Reply:
I strongly feel we should avoid rebase merges, not codify them as policy. Whilst not as bad as squash merges, they make it difficult to tell (without referring to GitHub) whether a commit is actually "finished" or "halfway through a PR". Yes, commits should be atomic so it shouldn't matter, but in reality they're not: if, say, a large staging-next PR contains a commit that breaks something, which is then fixed by a followup commit before the merge is done, it's not necessarily the casse that the PR will be modified to combine the breaking and fixing commits; therefore, it is useful to preserve in history "this entire set of changes we merged in was good, but halfway through the PR things might be broken".

Rebase merges also (aiui) make it impossible to use `git branch -d` to delete a branch iff it's been merged, since you lose any machine-readable relationship between "the commits you made" and "the commits that are on `master`".

See also nixpkgs-merge-bot which, afaict, just does "whatever the default is": https://github.com/NixOS/nixpkgs-merge-bot/blob/54faae5c11ae4fc092aabc37c76219fd233da973/nixpkgs_merge_bot/github/GitHubClient.py#L210-L212

Reply:
Dropped. I understand more and more the appeal of gerrits' changeset id and what jj/sapling is doing. Git simply doesn't have good mechanism for tracking commits that changes after code reviews.